content/manager/security.md

```diff
@@ -4,463 +4,741 @@ title: Security
+ True (or any non-empty value)
 
-## Secured Request Flow
+<br>
+### Web UI
+Credentials for all requests sent from the Web UI are set at Login. Do note that this client only supports
+authentication by username and password. Other authentication methods (e.g. tokens) are currently not supported.
 
-![request-flow]({{< img "guide/request_flow.png" >}})
+<br>
+### Cloudify's REST client
+Internally, Cloudify's CLI actually uses Cloudify's [REST client]
+(https://github.com/cloudify-cosmo/cloudify-rest-client/tree/3.3-build),
+which supports the variety of security configurations discussed in this guide. <br>To use it, create an instance of
+[CloudifyClient](https://github.com/cloudify-cosmo/cloudify-rest-client/blob/3.3/cloudify_rest_client/client.py#L231),
+initialized with values that match your configuration:
 
-## Internal communication between the Cloudify manager and other Cloudify components
+* `host` - The manager's host name of IP address. Default: *"localhost*"
+* `port` - The port to send requests to. Default: *80*
+* `protocol` - the protocol to send requests on. Default: *"http"*
+* `api_version` - version of REST API service to use. Default: *"v2"*
+* `headers` - header to send with each request. Default: *None*
+* `cert` - path to a client copy of the server's certificate (e.g. to validate self-signed certificates)
+* `trust_all` - if set to *True*, the server's certificate will be trusted without validation; if *False* - it will be
+validated. Default: *False*
 
-Currently, communication between the Cloudify agents and the Cloudify manager does not go through authentication - instead, REST calls from the agents to the manager are done to port 8101, which has the same general behavior as port 80, yet the REST service lets requests made to this port through without having them go through any of the security mechanisms.
 
-The usage of this port, however, is restricted to components on the same subnet as the Cloudify manager alone. This is done using specific security group rules, which are set up during bootstrap (for example, see [this rule in the Openstack manager blueprint](https://github.com/cloudify-cosmo/cloudify-manager-blueprints/blob/master/openstack/openstack-manager-blueprint.yaml#L206)). It is therefore impossible to bypass the security mechanisms from outside the manager's internal network by making REST calls directly to the 8101 port.
+<br>
+Here are two usage examples:
 
-In future versions, all communications from and to the Cloudify manager will utilize the security mechanisms, including communication with any of Cloudify's internal components, at which time, the bypass port 8101 will be removed.
+* Creating a REST client using credentials, to get a token from a secured manager **without SSL**
+{{< gsHighlight  python >}}
+from itsdangerous import base64_encode
+from cloudify_rest_client import CloudifyClient
 
+headers = {'Authorization': 'Basic ' + base64_encode('username:password')}
```

Comment:
I think wring MY_USERNAME:MY_PASSWORD would make it clearer these are user supplied values